### PR TITLE
[WIP] [CR] CRT Expansion

### PR DESCRIPTION
**Pull request by Soupster89**
```SUMMARY: Mods "Adds new core content around items, melee, mutations, and new professions"```

#### Purpose of change
To provide more content.

Martial arts and the like all work well with StatsWithSkills. Most of the content has pros and cons to provide balance or will be changed to reflect that, but I am mainly trying to get it to pass Travis as of now.

**Theme: Military/Sci-Fi/Dead Space**

_New Items:_
- guns, lasers with cons to counteract the lack of recoil
- gun mods, not that many yet, but they fit roles; don't use the tact-lite - it doesn't work properly-
- tool armor and armor, with pros and cons; quality but no storage, defensive but **heavy**, OP but encumbering and high volume 
- clothes to flesh out the new "C.R.I.T" group 
- tools, centered around portability at the cost of tool quality or other detriments
- weapons, some of them sap the life of enemies, others give speed boosts; artifact data infused gear

_Other:_
- mutation lines
- mutations
- professions
- rubber
- constructions
- martial arts & manuals, each with a boost based on stats and hit combos
- techniques, high move costs reduced by stat ratios, extra dmg based off of stats, stat ratio system ranking spoilers in the readme
- monsters from DS franchise

_Needs & WIP:_
- more recipes to create "advanced" gear; mid-game level stuff
- more groups for natural spawns of enemies and items
- more monsters
- balancing out armor values and weapons
- more fleshed out mutations and their trees
- less confusing descriptions that match CDDA verb-age
- squash random capitalizations
- more constructions, currently makeshift walls are kinda cool, but that's literally it
- scale back on some items, as survivor gear is similar to the C.R.I.T items

### CODE REVIEW

**Comment by Soupster89** [comment]
Fresh start. Too many conflicts to resolve and I had no real clue other than copy-paste strat.

**Comment by ifreund** [comment]
Since this looks pretty cool and large, I think it may make sense to squash bright nights and aftershock, as they are both unmaintained currently afaik and have a similar if not identical theme.

**Comment by Soupster89** [comment]
> Since this looks pretty cool and large, I think it may make sense to squash bright nights and aftershock, as they are both unmaintained currently afaik and have a similar if not identical theme.

I am honestly quite happy someone has finally taken the time to look through my stuff.

It has been over a few months now since I first started from just prying open a JSON file to change the Centipede Style, which then led to me creating more and more stuff, first going off of the subreddit and then finally being directed here to git.

**Comment by Fris0uman** [comment]
Not sure at all that aftershock is abandonned.

**Comment by kevingranade** [comment]
It has been explicitly abandoned by its author.


IF @soupster is interested in merging the content of BrightNights and/or Aftershock into their mod, that would be fine.  If they aren't that's fine, and we'll likely merge BrightNights and Aftershock at some point.

**Comment by Soupster89** [comment]
> It has been explicitly abandoned by its author.
> 
> IF @soupster is interested in merging the content of BrightNights and/or Aftershock into their mod, that would be fine. If they aren't that's fine, and we'll likely merge BrightNights and Aftershock at some point.

I would be delighted to! I honestly do not know what that really entails but I am very excited! How would I get about to doing just that?

**Comment by Naranar** [comment]
Poragon is already taking over Aftershock and I believe even if they weren't you'd need permission from the creator of aftershock to merge content from their mod.

**Comment by kevingranade** [comment]
I haven't seen any official statement from Poragon, but if they are great.
You do NOT need permission from mod authors (or anyone else, including me) to do anything you want with mod content, please do not spread misinformation.

Merging content would have two pieces.
1. Move the content into your mod directory.
2. Have the other mod declare a dependency on your mod in the modinfo.json file so that saves using that mod still have access to things originally added by it.

**Comment by Soupster89** [comment]
How do I go about making the checks pass? I currently see that there is something wrong about my terrain.json.

**Comment by AMurkin** [comment]
JSON files should be formatted with http://dev.narc.ro/cataclysm/format.html.

**Comment by Soupster89** [comment]
I went through and  linted everything.
Hopefully mobile doesn't mess

**Comment by Soupster89** [comment]
> JSON files should be formatted with http://dev.narc.ro/cataclysm/format.html.
> Don't forget line ending at the end of the files.

What is a line ending?

**Comment by AMurkin** [comment]
Last non-empty line should have a newline character.

Currently Jenkins complains about `data/mods/CRT_EXPANSION/martial/dragonslayertechn.json` formatting.
I think it's because newlines in that file are in windows format (CR LF).

**Comment by Soupster89** [comment]
> Last non-empty line should have a newline character.
> 
> Currently Jenkins complains about `data/mods/CRT_EXPANSION/martial/dragonslayertechn.json` formatting.
> I think it's because newlines in that file are in windows format (CR LF).

Alright, I'll look into that. I've used the validator/lint machine on the code for all of it doesn't look different.

It also shows errors in crt_recipes.json.

Also, where did you see that there was an error in dragonslayertechn.json? Is there a seperate window I am supposed to use?

I open uo travis and only see a make file error for my stuff.

**Comment by Soupster89** [comment]
Apparently "flat_bonus" did not get auto-formatted with the rest of the files, causing Travis to freak.

File: data/mods/CRT_EXPANSION/martial/dragonslayertechn.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  {
+    "type": "technique",
+    "id": "DSBI",
+    "name": "BISECTION",
+    "min_melee": 2,
+    "unarmed_allowed": true,
+    "melee_allowed": true,
+    "crit_tec": true,
+    "mult_bonuses": [ [ "movecost", 0.35 ], [ "damage", "bash", 1.05 ], [ "damage", "cut", 1.25 ], [ "damage", "stab", 1.05 ] ],
+    "flat_bonuses": [ [ "movecost", "str", -15 ], [ "movecost", "dex", -7 ], [ "damage", "bash", "per", 0.15 ], [ "damage", "cut", "per", 0.3 ]. [ "damage", "stab", "per", 0.15 ] ],
```
Comment (AMurkin) [inline data/mods/CRT_EXPANSION/martial/dragonslayertechn.json]:
```suggestion
    "flat_bonuses": [ [ "movecost", "str", -15 ], [ "movecost", "dex", -7 ], [ "damage", "bash", "per", 0.15 ], [ "damage", "cut", "per", 0.3 ], [ "damage", "stab", "per", 0.15 ] ],
```

Comment (Soupster89) [reply data/mods/CRT_EXPANSION/martial/dragonslayertechn.json]: Thank you so much!

**Comment by AMurkin** [comment]
I have the following error while loading a new world:
```
ERROR : src/mutation_data.cpp:181 [static void mutation_category_trait::check_consistency()] Mutation category WENDIGO has threshold mutation THRESH_WENDIGO, which does not exist
```

**Comment by Soupster89** [comment]
I will see to that, I haven't updated to the newest version yet, and mine is rather old.

I also need to upload the rest of the files

**Comment by Soupster89** [comment]
All of that went through lint too.

**Comment by ZhilkinSerg** [comment]
image

image

image

**Comment by Soupster89** [comment]
As of right now, I need to update the game itself to see what Zhilkin is commenting about.

I play/edit on android though so it will take a bit of time.

**Comment by ZhilkinSerg** [comment]
There are three files, which probably have invalid line endings:

image

**Comment by Soupster89** [comment]
> There are three files, which probably have invalid line endings:
> 
> image

All right, I'll get to it ASAP

**Comment by AMurkin** [comment]
Made a pull request into your fork.

File: data/mods/CRT_EXPANSION/modinfo.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+[
+  {
+    "type": "MOD_INFO",
+    "ident": "crt_expansion_git",
+    "name": "C.R.I.T Expansion Mod GIT EDITION",
```
Comment (ZhilkinSerg) [inline data/mods/CRT_EXPANSION/modinfo.json]: Are we really need `GIT EDITION` in mod name and `_git` in mod id?

**Comment by Soupster89** [comment]
I run it through the lint thing and it doesn't seem to produce the same results as yours.
:/

**Comment by Leosky** [comment]
Just for the note, i'm reworking book def and refactor the related code so you 'may' have to adapt a bit ma manuals but for now it 'should' be okay the old (crappy) way. Basicly, i will add a content section to book where you will be able to define a ma in a regular book without the current work around.

For exemple Aikodo manual will become something like :
```
{
    "id": "manual_aikido",
    "type": "BOOK",
    "name": "The Spirit of Aikido",
    "name_plural": "The Spirit of Aikido",
    "description": "A complete guide to Aikido.",
    "weight": 150,
    "volume": 1,
    "price": 20000,
    "material": "paper",
    "symbol": "?",
    "color": "green",
    "intelligence": 5,
    "time": "8 m",
    "content": {
      "fun_reading": {
        "fun": -1
      },
      "ma_training": {
        "martial_art": "aikido",
      }
    }
  }
```

or with alternative possible use of chapters : 

```
  {
    "id": "manual_aikido",
    "type": "BOOK",
    "name": "The Spirit of Aikido",
    "name_plural": "The Spirit of Aikido",
    "description": "A complete guide to Aikido.",
    "weight": 150,
    "volume": 1,
    "price": 20000,
    "material": "paper",
    "symbol": "?",
    "color": "green",
    "intelligence": 5,
    "skills_required":  [ "unarmed", 1],
    "time": "8 m",
    "content": {
      "fun_reading": {
        "fun": -1
      },
      "ma_training": {
        "martial_art": "aikido",
        "chapters": 15
      },
      "skill_training": {
        "skill": "unarmed",
        "max_level": 3,
        "chapters": 25
      }
    }
  },

```
Where you could also train unarmed combat for 25 times maximum, and learning akido after 15. "skills_required" and "chapters" becoming some kind of wild card conditional on reading event trigger.

**Comment by Soupster89** [comment]
> Just for the note, i'm reworking book def and refactor the related code so you 'may' have to adapt a bit ma manuals but for now it 'should' be okay the old (crappy) way. Basicly, i will add a content section to book where you will be able to define a ma in a regular book without the current work around.
> 
> For exemple Aikodo manual will become something like :
> 
> ```
> {
>     "id": "manual_aikido",
>     "type": "BOOK",
>     "name": "The Spirit of Aikido",
>     "name_plural": "The Spirit of Aikido",
>     "description": "A complete guide to Aikido.",
>     "weight": 150,
>     "volume": 1,
>     "price": 20000,
>     "material": "paper",
>     "symbol": "?",
>     "color": "green",
>     "intelligence": 5,
>     "time": "8 m",
>     "content": {
>       "fun_reading": {
>         "fun": -1
>       },
>       "ma_training": {
>         "martial_art": "aikido",
>       }
>     }
>   }
> ```
> 
> or with alternative possible use of chapters :
> 
> ```
>   {
>     "id": "manual_aikido",
>     "type": "BOOK",
>     "name": "The Spirit of Aikido",
>     "name_plural": "The Spirit of Aikido",
>     "description": "A complete guide to Aikido.",
>     "weight": 150,
>     "volume": 1,
>     "price": 20000,
>     "material": "paper",
>     "symbol": "?",
>     "color": "green",
>     "intelligence": 5,
>     "skills_required":  [ "unarmed", 1],
>     "time": "8 m",
>     "content": {
>       "fun_reading": {
>         "fun": -1
>       },
>       "ma_training": {
>         "martial_art": "aikido",
>         "chapters": 15
>       },
>       "skill_training": {
>         "skill": "unarmed",
>         "max_level": 3,
>         "chapters": 25
>       }
>     }
>   },
> ```
> 
> Where you could also train unarmed combat for 25 times maximum, and learning akido after 15. "skills_required" and "chapters" becoming some kind of wild card conditional on reading event trigger.

I see... I am still updating all of this anyways, most of my monsters won't spawn in just yet, the items need balancing and etc.

**Comment by Soupster89** [comment]
How do I update all of this? Do I make another PR request and maintain through that?

**Comment by Leosky** [comment]
you have time, i got little time to code atm and for it should be backward compatible.

**Comment by ZhilkinSerg** [comment]
Yes, if you need anything updated - you need to make new PR.

**Comment by kwl01skz** [comment]
C.R.I.T Tenacity have two identical bonuses.
"arpen", "stab", "dex",
"arpen", "stab", "dex",
It looks like a typo.

**Pull request merged**